ITL-7 Add models

BiroAron opened the pull request.

BiroAron committed:
ITL 31 - Add todo model

BiroAron committed:
ITL 32 - Add users model

tamasm-wg commented on server/models/todo.ts:
```diff
@@ -0,0 +1,24 @@
+  priority: TodoPriority;
+  description?: String;
+  isChecked: boolean;
+  date: String;
```
shouldn't the date type be Date ?

BiroAron replied:
In the front-end of the application, the values of the date are set via an input field with the type of date wich gets the date in the format of dd-mm-yyyy in a string, so thats why im storing it as a string.

SzaszWebgurus replied:
use Date here as type

tamasm-wg commented on server/models/todo.ts:
```diff
@@ -0,0 +1,24 @@
+  priority: { type: String, enum: ["High", "Medium", "Low"] },
+  description: String,
+  isChecked: Boolean,
+  date: String,
```
here also

SzaszWebgurus replied:
same as above

SzaszWebgurus commented on server/models/todo.ts:
```diff
@@ -0,0 +1,24 @@
+  date: String,
+});
+
+const todo: Model<ITodo> =
```
you dont need a variable here, you can export it here

SzaszWebgurus commented on server/models/user.ts:
```diff
@@ -0,0 +1,18 @@
+  password: { type: String, required: true },
+});
+
+const user: Model<IUser> =
```
same as above

BiroAron committed:
ITL 7 - Made changes regarding the comments

-changed the type of date in the todos model
-exported without creating variables
